### PR TITLE
fix: set `lastUpdated` time to git commit timestamp

### LINKED ISSUE #343: Last updated time error
### Describe the bug

Each time I build docs with GitHub Action (workflows), every page generated by VitePress set its 'Last Updated' time up to date.

### Reproduction

See [Demo](https://tazimi.dev/awesome-notes/).

### Expected behavior

'Last Update' time should be consistent to git update timestamp, shouldn't consistent to `vitepress build` timestamp.

### System Info

```shell
System:
    OS: Linux 5.8 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 2.71 GB / 7.66 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 16.4.2 - ~/.nvm/versions/node/v16.4.2/bin/node
    npm: 7.18.1 - ~/.nvm/versions/node/v16.4.2/bin/npm
  Browsers:
    Firefox: 90.0
  npmPackages:
    vitepress: ^0.15.6 => 0.15.6
```

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/coc)
- [X] Read the [docs](https://vitepress.vuejs.org/).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by sabertazimi**
More details on issue #343

close #343

### CODE REVIEW

File: src/client/theme-default/components/LastUpdated.vue [hunk 1]
```diff
@@ -16,11 +16,16 @@ const prefix = computed(() => {
...
 const datetime = ref('')
+
 onMounted(() => {
   // locale string might be different based on end user
   // and will lead to potential hydration mismatch if calculated at build time
   datetime.value = new Date(page.value.lastUpdated).toLocaleString('en-US')
 })
+
+onUpdated(() => {
+  datetime.value = new Date(page.value.lastUpdated).toLocaleString('en-US')
```
Comment (posva) [inline src/client/theme-default/components/LastUpdated.vue:27]: Why is this necessary? It's already done in onMounted

Comment (sabertazimi) [reply src/client/theme-default/components/LastUpdated.vue:27]: When route changing, without `onUpdated` hook leading to the same last updated time on every page.

Cause I'm not familar with Vue, I don't know why it happened.

**Review by sabertazimi — COMMENTED** [review]
Closed.
